engine/src/main/scala/cromwell/engine/workflow/lifecycle/CopyWorkflowOutputsActor.scala

```diff
@@ -0,0 +1,64 @@
+package cromwell.engine.workflow.lifecycle
+
+import java.nio.file.Path
+
+import akka.actor.Props
+import cromwell.core.{PathCopier, WorkflowId}
+import cromwell.engine.EngineWorkflowDescriptor
+import cromwell.engine.backend.{BackendConfiguration, CromwellBackends}
+import cromwell.engine.workflow.lifecycle.execution.{ExecutionStore, OutputStore}
+import wdl4s.ReportableSymbol
+import wdl4s.values.WdlSingleFile
+
+object CopyWorkflowOutputsActor {
+  def props(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor, executionStore: ExecutionStore,
+            outputStore: OutputStore) = Props(
+    new CopyWorkflowOutputsActor(workflowId, workflowDescriptor, executionStore, outputStore)
+  )
+}
+
+class CopyWorkflowOutputsActor(workflowId: WorkflowId, val workflowDescriptor: EngineWorkflowDescriptor,
+                               executionStore: ExecutionStore, outputStore: OutputStore)
+  extends EngineWorkflowCopyFinalizationActor {
+
+  override def copyFiles(): Unit = {
+    getWorkflowOption("outputs_path") foreach copyWorkflowOutputs
+  }
+
+  private def copyWorkflowOutputs(workflowOutputsFilePath: String): Unit = {
+    val workflowOutputsPath = convertStringToPath(workflowOutputsFilePath)
+
+    val reportableOutputs = workflowDescriptor.backendDescriptor.workflowNamespace.workflow.outputs
+
+    val outputFilePaths = getOutputFilePaths(reportableOutputs)
+
+    outputFilePaths foreach {
+      case (workflowRootPath, srcPath) =>
+        PathCopier.copy(workflowRootPath, srcPath, workflowOutputsPath)
+    }
+  }
+
+  private def getOutputFilePaths(reportableOutputs: Seq[ReportableSymbol]): Seq[(Path, Path)] = {
+    val results = for {
+      reportableOutput <- reportableOutputs
+      // NOTE: Without .toSeq, output in arrays only yield the last output
+      (backend, calls) <- workflowDescriptor.backendAssignments.groupBy(_._2).mapValues(_.keys.toSeq).toSeq
+      config <- BackendConfiguration.backendConfigurationDescriptor(backend).toOption.toSeq
+      rootPath <- CromwellBackends.shadowBackendLifecycleFactory(backend).toOption. // Try/Option/Seq soup
```

Comment:
Would it be possible to pull the rhs of this one into a function w/ a descriptive name? I'm thinking that it's a 2 line rhs which requires a comment, it makes it tough when parsing all the steps of the forcomp